From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 17 Nov 2020 14:39:59 +0100
Subject: clutter/click-action: Move away from device IDs

Store the press device itself, not its ID.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1569>
---
 clutter/clutter/clutter-click-action.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/clutter/clutter/clutter-click-action.c b/clutter/clutter/clutter-click-action.c
index 6bc977a..570d86d 100644
--- a/clutter/clutter/clutter-click-action.c
+++ b/clutter/clutter/clutter-click-action.c
@@ -114,7 +114,7 @@ struct _ClutterClickActionPrivate
   gint drag_threshold;
 
   guint press_button;
-  gint press_device_id;
+  ClutterInputDevice *press_device;
   ClutterEventSequence *press_sequence;
   ClutterModifierType modifier_state;
   gfloat press_x;
@@ -299,7 +299,7 @@ on_event (ClutterActor       *actor,
         return CLUTTER_EVENT_PROPAGATE;
 
       priv->press_button = has_button ? clutter_event_get_button (event) : 0;
-      priv->press_device_id = clutter_event_get_device_id (event);
+      priv->press_device = clutter_event_get_device (event);
       priv->press_sequence = clutter_event_get_event_sequence (event);
       priv->modifier_state = clutter_event_get_state (event);
       clutter_event_get_coords (event, &priv->press_x, &priv->press_y);
@@ -370,7 +370,7 @@ on_captured_event (ClutterActor       *stage,
 
       if ((has_button && clutter_event_get_button (event) != priv->press_button) ||
           (has_button && clutter_event_get_click_count (event) != 1) ||
-          clutter_event_get_device_id (event) != priv->press_device_id ||
+          clutter_event_get_device (event) != priv->press_device ||
           clutter_event_get_event_sequence (event) != priv->press_sequence)
         return CLUTTER_EVENT_PROPAGATE;
 
@@ -411,7 +411,7 @@ on_captured_event (ClutterActor       *stage,
         gfloat motion_x, motion_y;
         gfloat delta_x, delta_y;
 
-        if (clutter_event_get_device_id (event) != priv->press_device_id ||
+        if (clutter_event_get_device (event) != priv->press_device ||
             clutter_event_get_event_sequence (event) != priv->press_sequence)
           return CLUTTER_EVENT_PROPAGATE;
 
